### PR TITLE
[RW-528] Don't double-create accounts, don't check for username ''

### PR DESCRIPTION
**Pull request by calbach**
- We were submitting both on form submit and click, so one would generally fail with a 409
- We were doing username checks on empty string, which was a 403 error
- Add regression test
- Clean up dead properties

### CODE REVIEW

File: ui/src/app/views/account-creation/component.ts [hunk 1]
```diff
@@ -99,6 +99,9 @@ export class AccountCreationComponent {
     this.conflictError = false;
     clearTimeout(this.usernameCheckTimeout);
     this.usernameCheckTimeout = setTimeout(() => {
+      if (!this.profile.username) {
```
Comment (johnstill) [inline ui/src/app/views/account-creation/component.ts]: Maybe pop a `.trim()` on `username` as well b/c:

```
if ('  ') { console.log('hello') } // -> 'hello'
```

File: ui/src/app/views/account-creation/component.html [hunk 2]
```diff
@@ -29,10 +29,12 @@ <h5 style="display: inline-block;">@{{gsuiteDomain}}</h5>
...
-      <button (click)="createAccount()" class="btn btn-primary short-button">Next</button>
-      <span class="spinner spinner-inline" *ngIf="creatingAcccount">
-        Creating account...
-      </span>
+      <button type="submit"
+          class="btn btn-primary short-button"
+          [disabled]="creatingAcccount">
+        Next
+      </button>
+      <span class="spinner spinner-inline" *ngIf="creatingAcccount"></span>
```
Comment (johnstill) [inline ui/src/app/views/account-creation/component.html]: I'd put `[clrLoading]="creatingAccount"` on the `<button>`... not sure how well that works anymore but it usually looks pretty clean.  You can [see it in action](https://vmware.github.io/clarity/documentation/v0.11/buttons) (bottom of the page).

File: ui/src/app/views/account-creation/component.ts [hunk 3]
```diff
@@ -43,6 +40,9 @@ export class AccountCreationComponent {
   gsuiteDomain: string;
   usernameCheckTimeout: NodeJS.Timer;
 
+  // TODO: Injecting the parent component is a bad separation of concerns, as
+  // well as injecting LoginComponent. Should look at refactoring these
+  // interactions.
```
Comment (johnstill) [inline ui/src/app/views/account-creation/component.ts]: :100:

That we can do this at all is interesting, but it makes me nervous.  I don't _think_ this is an intended use case (hard to find any docs or other info on it); I think we're operating outside of Angular's support by doing this.

File: ui/src/app/views/account-creation/component.html [hunk 4]
```diff
@@ -32,10 +32,11 @@ <h5 style="display: inline-block;">@{{gsuiteDomain}}</h5>
     </div>
 
     <div class="form-section">
-      <button (click)="createAccount()" class="btn btn-primary short-button">Next</button>
-      <span class="spinner spinner-inline" *ngIf="creatingAcccount">
-        Creating account...
-      </span>
+      <button type="submit"
+          class="btn btn-primary short-button"
+          [clrLoading]="creatingAcccount">
```
Comment (blrubenstein) [inline ui/src/app/views/account-creation/component.html]: Nit: Spelling

File: ui/src/app/views/account-creation/component.html (hunk 2 above)
Comment (calbach) [reply ui/src/app/views/account-creation/component.html]: Good call - done.

File: ui/src/app/views/account-creation/component.html [hunk 5]
```diff
@@ -32,10 +32,12 @@ <h5 style="display: inline-block;">@{{gsuiteDomain}}</h5>
...
 
     <div class="form-section">
-      <button (click)="createAccount()" class="btn btn-primary short-button">Next</button>
-      <span class="spinner spinner-inline" *ngIf="creatingAcccount">
-        Creating account...
-      </span>
+      <button type="submit"
+          class="btn btn-primary short-button"
+          [disabled]="creatingAcccount"
+          [clrLoading]="creatingAcccount">
```
Comment (blrubenstein) [inline ui/src/app/views/account-creation/component.html]: Nit: Spelling

File: ui/src/app/views/account-creation/component.ts (hunk 1 above)
Comment (calbach) [reply ui/src/app/views/account-creation/component.ts]: Done.

File: ui/src/app/views/account-creation/component.html (hunk 5 above)
Comment (calbach) [reply ui/src/app/views/account-creation/component.html]: Huh - good catch. Incredibly this was mispelled everywhere so it actually was working - likely the work of some autocomplete/IDE.

**Pull request merged**